src/util/serverUtils.js

```diff
@@ -128,31 +119,16 @@ export function registerExtensionEvents(server) {
 }
 
 /**
- * Make any environment changes that need to be made in response to the provided
- * config
- * @param {Object} config
- * @return {Object} the original config object
- */
-export function configureEnv(config) {
-	// When using .dev.meetup endpoints, ignore self-signed SSL cert
-	const USING_DEV_ENDPOINTS = checkForDevUrl(config);
-	https.globalAgent.options.rejectUnauthorized = !USING_DEV_ENDPOINTS;
-
-	return config;
-}
-
-/**
  * server-starting function
  */
-export function server(routes, connection, plugins, platform_agent, config) {
+export function server(routes, connection, plugins, platform_agent) {
 	const server = new Hapi.Server();
 
 	// store runtime state
 	// https://hapijs.com/api#serverapp
-	server.app = {
-		isDevConfig: checkForDevUrl(config),  // indicates dev API or prod API
-		...config
-	};
+	// TODO: do we still need this?
```

Comment:
Yes, and in general we should try to exclusively use the `server.app` object to refer to server configuration - it's a [Hapi convention](https://hapijs.com/api#serverapp) that reduces the need for importing config ad-hoc. I like the `server.app` property because it forces "server configuration" to _be part of the server_ object. Config that is _not_ related to the server can go elsewhere, but we might not have anything else.

I've noted a few places where `server.app` can/should be used in this PR instead of importing `config`, so hopefully that can reduce the need for ad-hoc config imports, which can make the dependency structure harder to see and maintain. It might take a few iterations to fully clean it up (my ideal is that we import `config` exactly once - when the server is first configured - but we'll see how close we can get)